problems/arraysWithLoops.js

```diff
@@ -25,39 +38,82 @@ function shoutWhileLoop() {}
  * @returns {Array} - ["I!", "am!", "a!", "happy!", "array!"]
  */
 
-function shoutForOfLoop() {}
+function shoutForOfLoop(array) {
+  i = 0
+  for (element of array){
+    i < array.length
+    array[i] = array[i] + "!";
+    i++
+
+  }
+
+  return array
+}
 
 /**
  * Takes in an array and returns the sum of all values
  * @param {number[]} nums
  * @returns {number} sum
  */
 
-function sumArray() {}
+function sumArray(nums) {
+  sum = 0
+  var i;
+  var sum = 0;
+  for (i = 0; i < nums.length; i++) {
+    sum += nums[i];
+  }
+  return sum;
+}
 
 /**
  * Takes in an array and returns an array of all the odd valued elements
  * @param {number[]} nums
  * @returns {array} odds
  */
 
-function oddArray() {}
+function oddArray(nums) {
+  let array = []
+  for (let element of nums){
+    if (element % 2 === 1){
+      array.push(element)
+    }
+  }
+  return array
+}
 
 /**
  * Takes in an array and returns an array of all the even valued elements
  * @param {number[]} nums
  * @returns {array} evens
  */
 
-function evenArray() {}
+function evenArray(nums) {
+  let evenNum = []
+  for (let element of nums){
+    if (element % 2 === 0){
+      evenNum.push(element)
+    }
+  }
+  return evenNum
+}
+
 
 /**
  * Takes in array and returns the smallest number in the array
  * @param {number[]} nums
  * @returns {number} smallest value
  */
 
-function findSmallest() {}
+function findSmallest(nums) {
+  let i = nums[0]
```

Comment:
you should name your variables specifically for what you are using. This should be "smallNum" or something indicative of what the variable is used for